index.js

```diff
@@ -91,37 +97,41 @@ function TChannel(options) {
     self.serverSocket.on('close', function onServerSocketClose() {
         self.logger.warn('server socket close');
     });
-
-    if (self.listening) {
-        self.listen();
-    }
 }
 require('util').inherits(TChannel, require('events').EventEmitter);
 
 // Decoulping config and creation from the constructor.
-// This also allows us to better unit test the code as the test process
-// is not blocked by the listening connections
-TChannel.prototype.listen = function listen() {
+TChannel.prototype.listen = function listen(port, host, callback) {
     var self = this;
-    if (!self.serverSocket) {
-        throw new Error('Missing server Socket.');
+    if (self.listened) {
+        throw new Error('TChannel can only listen once');
     }
-    if (!self.host) {
-        throw new Error('Missing server host.');
+    if (typeof host !== 'string') {
+        throw new Error('TChannel requires host argument');
     }
-    if (!self.port) {
-        throw new Error('Missing server port.');
+    if (host === '0.0.0.0') {
+        throw new Error('TChannel must listen with externally visible host');
     }
-
-    self.serverSocket.listen(self.port, self.host);
+    // Default port to 4040 if null or undefined. 0 is a valid port, indicating
+    // that the system should assign an available port.
+    port = port === undefined || port === null ? 4040 : port;
```

Comment:
can we remove the default port in this pr as well. having a default port is something we wanted to remove .